int32 direction
float32 radius
---
bool success